### PR TITLE
Update telemetry docs for 5.22

### PR DESCRIPTION
**Pull request by jasonblais**
@wiersgallak @amyblais @mkraft Please see PR comments on the specific sections I'm asking for your review

What this PR does:

1. Add diagnostics for channel moderation https://github.com/mattermost/mattermost-server/pull/14095 and groups https://github.com/mattermost/mattermost-server/pull/13736
2. Add new diagnostics added for config settings
3. Fix some diagnostics incorrectly marked as client-side (permissions), moving them to server-side telemetry
4. Add a note that most client-side telemetry has been disabled since Mattermost v5.8, referencing the source file for latest list of client-side events sent via telemetry

### CODE REVIEW

File: source/administration/telemetry.rst [hunk 1]
```diff
@@ -56,15 +56,27 @@ Server Configuration Settings
 Commercial License Information (Enterprise Edition Only)
   Information about commercial license key purchased or trial license key used for Enterprise Edition servers: Company ID, license ID, license issue date, license start date, license expiry date, number of licensed users, license short name (E10 vs E20), list of unlocked Enterprise features.
 
+Channel Moderation Configuration Information (Enterprise Edition Only)
```
Comment (jasonblais) [inline source/administration/telemetry.rst:59]: @mkraft Please review channel moderation telemetry for correctness (from https://github.com/mattermost/mattermost-server/pull/14095)

File: source/administration/telemetry.rst [hunk 2]
```diff
@@ -56,15 +56,27 @@ Server Configuration Settings
...
+  Information related to channel moderation, including number of channel schemes, number of channels with posting messages disabled for users or guests, number of channels with emoji reactions disabled for users or guests, number of channels with managing members disabled, number of channels with channel mentions disabled for users or guests.
+
+Groups Configuration Information (Enterprise Edition Only)
```
Comment (jasonblais) [inline source/administration/telemetry.rst:62]: @mkraft Please review groups telemetry for correctness (from https://github.com/mattermost/mattermost-server/pull/13736)

File: source/administration/telemetry.rst (hunk 1 above)
Comment (jasonblais) [inline source/administration/telemetry.rst:59]: @amyblais please only editor review changes from here onward. The change to config settings above this does not need editor review.

File: source/administration/telemetry.rst [hunk 3]
```diff
@@ -56,15 +56,27 @@ Server Configuration Settings
...
+Permissions Configuration Information (Enterprise Edition Only)
+  Permissions configured for each role for the System Scheme and each Team Override Scheme created in the system. Scheme ID; Team Admin Permissions; Team User Permissions; Channel Admin Permissions; Channel User Permissions; Number of teams the scheme is associated with
+
 Aggregated Usage Statistics
   Non-personally identifiable summations of basic usage statistics: Number of enabled and disabled accounts, number of user logins in the last 24 hours and the last 30 days, number of users active in the last day/month, whether APIv3 endpoints were used in the last 24 hours, number of posts, channels, teams and bots.
 
 The following information is sent when the specified event occurs:
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+  .. note::
```
Comment (jasonblais) [inline source/administration/telemetry.rst:77]: @wiersgallak Please review this note

File: source/administration/telemetry.rst (hunk 2 above)
Comment (mkraft) [reply source/administration/telemetry.rst:62]: @jasonblais LGTM 👍

**Pull request merged**